tiatoolbox/visualization/bokeh_app/main.py

```diff
@@ -0,0 +1,1944 @@
+        # its a graph
+        handle_graph_layer(attr)
+        return
+
+    # otherwise its a tile-based overlay of some form
+    fname = make_safe_name(attr.item)
+    resp = UI["s"].put(
+        f"http://{host2}:5000/tileserver/overlay",
+        data={"overlay_path": fname},
+    )
+    resp = json.loads(resp.text)
+
+    if Path(attr.item).suffix in [".db", ".dat", ".geojson"]:
+        update_ui_on_new_annotations(resp)
+    else:
+        add_layer(resp)
+        change_tiles(resp)
+
+
+def layer_select_cb(attr: ButtonClick) -> None:  # noqa: ARG001
+    """Callback to handle toggling specific annotation types on and off."""
+    build_predicate()
+    UI["vstate"].update_state = 1
+    UI["vstate"].to_update.update(["overlay"])
+
+
+def fixed_layer_select_cb(obj: Button, attr: ButtonClick) -> None:  # noqa: ARG001
+    """Callback to handle toggling non-annotation layers on and off."""
+    key = UI["vstate"].layer_dict[obj.label]
+    if obj.label == "edges":
+        if not UI["p"].renderers[key].visible:
+            UI["p"].renderers[key].visible = True
+        else:
+            UI["p"].renderers[key].visible = False
+    elif obj.label == "nodes":
+        if UI["p"].renderers[key].glyph.fill_alpha == 0:
+            UI["p"].renderers[key].glyph.fill_alpha = UI["overlay_alpha"].value
+            UI["p"].renderers[key].glyph.line_alpha = UI["overlay_alpha"].value
+        else:
+            UI["p"].renderers[key].glyph.fill_alpha = 0.0
+            UI["p"].renderers[key].glyph.line_alpha = 0.0
+    elif UI["p"].renderers[key].alpha == 0:
+        UI["p"].renderers[key].alpha = float(obj.name)
+    else:
+        obj.name = str(UI["p"].renderers[key].alpha)  # save old alpha
+        UI["p"].renderers[key].alpha = 0.0
+
+
+def layer_slider_cb(
+    obj: Slider,
+    attr: str,  # noqa: ARG001
+    old: float,  # noqa: ARG001
+    new: float,
+) -> None:
+    """Callback to handle changing the alpha of a layer."""
+    if obj.name.split("_")[0] == "nodes":
+        set_alpha_glyph(
+            UI["p"].renderers[UI["vstate"].layer_dict[obj.name.split("_")[0]]].glyph,
+            new,
+        )
+    elif obj.name.split("_")[0] == "edges":
+        UI["p"].renderers[
+            UI["vstate"].layer_dict[obj.name.split("_")[0]]
+        ].glyph.line_alpha = new
+    else:
+        UI["p"].renderers[UI["vstate"].layer_dict[obj.name.split("_")[0]]].alpha = new
+
+
+def color_input_cb(
+    obj: ColorPicker,
+    attr: str,  # noqa: ARG001
+    old: str,  # noqa: ARG001
+    new: str,
+) -> None:
+    """Callback to handle changing the color of an annotation type."""
+    UI["vstate"].mapper[UI["vstate"].orig_types[obj.name]] = (*hex2rgb(new), 1)
+    if UI["vstate"].cprop == "type":
+        update_renderer("mapper", UI["vstate"].mapper)
+    UI["vstate"].update_state = 1
+    UI["vstate"].to_update.update(["overlay"])
+
+
+def bind_cb_obj(cb_obj: Model, cb: Callable[[Model, str, Any, Any]]) -> Callable:
```

Comment:
I am getting the following error on this line.

'Callable' must be used as 'Callable[[arg, ...], result]'

Reply:
This stuff is getting seriously frustrating.

It's happy with Callable[[Model, str, Any, Any]] on my computer, and on github actions (otherwise everything wouldn't have passed), why is it demanding the return type also on your machine? Do you have a different version of something set up?

Reply:
All this needs is the result type of the Callable. e.g., you can update this to 
```suggestion
def bind_cb_obj(cb_obj: Model, cb: Callable[[Model, str, Any, Any], None]) -> Callable:
```

Reply:
Here are the details:
https://peps.python.org/pep-0677/
This allows types like Callable[[int, str], bool] to be written as (int, str) -> bool.

Reply:
fixing it is not hard, it just takes time and it is frustrating to be chasing an ever-moving target with respect to what these checks require